src/app/models/main.rs

```diff
@@ -99,6 +109,12 @@ impl AlbumDescription {
         let duration: u32 = self.songs.iter().map(|song| song.duration).sum();
         format_duration(duration.into())
     }
+    pub fn year(&self) -> Option<u32> {
+        match self.release_date {
+            Some(ref date) => Some(u32::from_str(date.split('-').next().unwrap()).unwrap()),
```

Comment:
it is a bit unfortunate to unwrap and wrap it back up!

you could probably handle this a little bit better using [map ](https://doc.rust-lang.org/std/option/enum.Option.html#method.map)or [and_then](https://doc.rust-lang.org/std/option/enum.Option.html#method.and_then), and [ok](https://doc.rust-lang.org/std/result/enum.Result.html#method.ok)

Reply:
Ah, yes, that is exactly I had to use those functions. In the latest commit I'm using `and_then` and `ok` :)